docs/reference/migration/apis/upgrade.asciidoc

```diff
@@ -6,8 +6,14 @@
 <titleabbrev>Migration upgrade</titleabbrev>
 ++++
 
-The Migration Upgrade API performs the upgrade of internal indices to make them
-compatible with the next major version.
+IMPORTANT: To prepare a cluster for an upgrade to Elasticsearch version 7, use
+the {kibana-ref}/upgrade-assistant.html[{kib} Upgrade Assistant] or
+<<reindex-upgrade,Reindex manually>> before upgrading.
+
+deprecated[6.7, Use the {kibana-ref}/upgrade-assistant.html[{kib} Upgrade
+Assistant] or <<reindex-upgrade,Reindex manually>> before upgrading.] The
+Migration Upgrade API performs the upgrade of internal indices to make them
+compatible with version 6 of Elasticsearch.
```

Comment:
If someone uses this API in 6.7 will it work correctly ?

Reply:
I've made a tweak to this to make sure the behavior aligns, but as of this PR, if used in 6.7, the Upgrade API will only upgrade indices that need to be upgraded from their 5.x format (i.e. they were not upgraded before upgrading the cluster). Otherwise, it will just say that the index needs to be reindexed.